### PR TITLE
add PrimeField::char method

### PR DESCRIPTION
**Pull request by bchyl**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -215,6 +215,9 @@ pub trait PrimeField: Field + From<u64> {
         !self.is_odd()
     }
 
+    /// Returns the field characteristic; the modulus.
+    fn char() -> Self::Repr;
```
Comment (str4d) [inline src/lib.rs:219]: What use case do you have for this API, that isn't served by `PrimeFieldBits::char_le_bits`?

**Comment by tarcieri** [comment]
Isn't this the sort of value you don't want to allow to "escape" into the public API because technically it's invalid? (i.e. field elements are exclusive of the modulus)

**Comment by str4d** [comment]
Yeah, though note that this is returning a `PrimeField::Repr` which is a byte encoding, so while that is technically a type error (as the characteristic can never be a field element representation), it isn't as bad as returning a `PrimeField` instance.

We also already expose the modulus via `PrimeFieldBits::char_le_bits`, but the only place I'm aware of that being used is inside `bellman`'s boolean gadget, and there only so we can drop the leading zeroes in the bit representation of the field element (to handle cases where the bit representation is larger than `PrimeField::NUM_BITS`).

I am currently -1 on merging another way to expose the modulus without having a use case.

**Comment by str4d** [comment]
In a discussion with @ebfull, we decided that we would not merge this PR as-is, because `PrimeField::Repr` is semantically intended to be a field element representative which the modulus will never be. Separately, we are going to add a `const MODULUS: &'static str` associated constant to `PrimeField` for debugging purposes, but that is treated as opaque (so if you are using a particular field concretely then you could parse it to obtain the modulus, but in generic code you won't know its encoding).

**Pull request closed without merging**